jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererUtil.kt

```diff
@@ -36,6 +37,16 @@ import software.aws.toolkits.jetbrains.utils.notifyWarn
 import software.aws.toolkits.resources.message
 import software.aws.toolkits.telemetry.CodewhispererCompletionType
 
+fun runIfIamIdentityCenterConnection(project: Project, callback: (connection: ToolkitConnection) -> Any?) =
```

Comment:
sorry @andrewyuq , I was thinking to make callback to return `T?` like `runReadAction`, when we don't need a return value, we simply don't use it. I think if we want to make `Unit` a part of this callback, the origianl plan is good enough, let's keep it `fun runIf(): Unit`. And we will add a `fun calculateIf(): T?` when it's needed

Reply:
think it's unsafe to use Any unless it's required as we need lot of casting and check afterward

Reply:
yea, I have changed back to Unit for now.

Reply:
gotcha